SIPS/sip-32.md

```diff
@@ -0,0 +1,89 @@
+---
+sip: 32
+title: Add `snap_trackEvent` method for pre-installed Snaps
+status: Draft
+author: Daniel Rocha (@danroc)
+created: 2025-03-20
+---
+
+## Abstract
+
+This Snap Improvement Proposal (SIP) introduces a new method,
+`snap_trackEvent`, allowing pre-installed Snaps to submit tracking events
+through the client. The client will determine how to handle these events.
+
+This feature enables Snaps to utilize the existing MetaMask client
+infrastructure for event tracking while maintaining user privacy.
+
+## Motivation
+
+Currently, there is no standardized way for pre-installed Snaps to submit
+analytics or tracking events. This proposal aims to:
+
+- Enable pre-installed Snaps to reuse MetaMask’s event tracking infrastructure.
+
+- Allow the MetaMask client to control event handling, ensuring security and
+  compliance with privacy policies.
+
+## Specification
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### `snap_trackEvent`
+
+The `snap_trackEvent` method allows pre-installed Snaps to submit structured
+tracking events to the client. It is the client’s responsibility to process
+these events accordingly.
+
+The client MUST ensure that only pre-installed Snaps can call this method.
+
+The `snap_trackEvent` method is defined as follows:
+
+#### Parameters
+
+- `event` - Event object to be tracked.
+  - `category: string` - The category to associate the event.
+
+  - `name: string` - The name of the event to track.
+
+  - `properties: Record<string, Json>` - (**Optional**) Custom values to track.
+    The client MUST enforce that all keys in this object are in the
+    `snake_case` format.
```

Comment:
Is this actually a requirement for our analytics, or simply a convention?

Reply:
The data team might want to enforce it in the future, but currently it's only a convention